crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -177,45 +83,67 @@ pub struct CheckpointExecutorEventLoop {
     /// the final checkpoint will not have been set, thus CheckpointExecutor
     /// will reschedule the last checkpoint and correctly set end_of_epoch.
     end_of_epoch: bool,
+    // If true, need to run crash recovery
+    cold_start: bool,
```

Comment:
I am not sure how `cold_start` works. Could you comment more?
We don't persist its value to disk, so we always attempt to do crash recovery, right?

Reply:
So this new version of the PR moves the instantiation of `CheckpointExecutor` outside of the loop. Therefore, `cold_start` is a flag that starts as `false` and is set to `true` once we've done crash recovery. That way when we re-run executor (for the next epoch) we do not rerun crash recovery. It will only re-run if we re-instantiate